src/backend/distributed/planner/multi_logical_planner.c

```diff
@@ -3531,12 +3542,50 @@ ApplyLocalJoin(MultiNode *leftNode, MultiNode *rightNode,
 }
 
 
+/*
+ * ApplySingleRangePartitionJoin is a wrapper around ApplySinglePartitionJoin()
+ * which sets the joinRuleType properly.
+ */
+static MultiNode *
+ApplySingleRangePartitionJoin(MultiNode *leftNode, MultiNode *rightNode,
+							  Var *partitionColumn, JoinType joinType,
+							  List *applicableJoinClauses)
+{
+	MultiJoin *joinNode =
+		ApplySinglePartitionJoin(leftNode, rightNode, partitionColumn, joinType,
+								 applicableJoinClauses);
+
+	joinNode->joinRuleType = SINGLE_RANGE_PARTITION_JOIN;
+
+	return (MultiNode *) joinNode;
+}
+
+
+/*
+ * ApplySingleHashPartitionJoin is a wrapper around ApplySinglePartitionJoin()
+ * which sets the joinRuleType properly.
+ */
+static MultiNode *
+ApplySingleHashPartitionJoin(MultiNode *leftNode, MultiNode *rightNode,
+							 Var *partitionColumn, JoinType joinType,
+							 List *applicableJoinClauses)
+{
+	MultiJoin *joinNode =
+		ApplySinglePartitionJoin(leftNode, rightNode, partitionColumn, joinType,
+								 applicableJoinClauses);
+
+	joinNode->joinRuleType = SINGLE_HASH_PARTITION_JOIN;
```

Comment:
would it make sense for this to become a parameter?

Reply:
I'd have a slight preference for that

Reply:
Ok, I remembered the reason now :) 

It'd break the `RuleApplyFunction()` signature. So, do you still prefer that if I need to change it?